### ISSUE TITLE
Rename `Update` to `StoreInSalsaDb` or something like that

### ISSUE THREAD

**Issue by nikomatsakis** [salsa-rs/salsa#592]
The name `Update` is highly unintuitive. What this value really means is "safe to store in the salsa database".

**Comment by puuuuh** [comment]
SalsaIngredient?

**Comment by nikomatsakis** [comment]
Yeah, it is not an ingredient. It is for data structures that can be included in tracked/interned structs but which are not, themselves, tracked/interned structs.

**Comment by nikomatsakis** [comment]
What about `SalsaStructMember` ?

**Comment by MichaReiser** [comment]
My understanding of the `Update` trait is still very limited but I like the name if it is the trait that must be implemented by every salsa-struct field (maybe  `SalsaStructField`?)

**Comment by nikomatsakis** [comment]
I like the name `SalsaStructField`. Also, as discussed in our sync meeting today, I wonder if we should remove the `'static` fallback (or,  more likely, make it "opt-in" with an annotation on the field). The intent would be that deriving `SalsaStructField` is more efficient, and it also would make it possible for us to discourage people from using `HashMap` and `HashSet` (as opposed to `FxHashSet` etc) since those can easily lead to non-deterministic results.

**Comment by MichaReiser** [comment]
> The intent would be that deriving SalsaStructField is more efficient

I'm open to trying this on Ruff if we have a PR that adds this requirement to get some data on *how painful* it is for users.

**Comment by MichaReiser** [comment]
I'm leaning towards calling this `SalsaValue`. It's a value that you can store in a salsa db. Whether this is as part of a tracked, interned, or input struct or as a query return value.

Splitting the traits also has the benefit that `Update`, the trait that you may need to implement manually, doesn't have to be unsafe anymore.
